akka-persistence-typed/src/main/scala/akka/persistence/typed/internal/EventsourcedRunning.scala

```diff
@@ -257,14 +258,6 @@ private[akka] object EventsourcedRunning {
         event.getClass.getName, seqNr, setup.persistenceId, cause.getMessage)
     }
 
-    private def onPersistFailureThenStop(cause: Throwable, event: Any, seqNr: Long): Behavior[InternalProtocol] = {
-      setup.log.error(cause, "Failed to persist event type [{}] with sequence number [{}] for persistenceId [{}].",
-        event.getClass.getName, seqNr, setup.persistenceId)
-
-      // FIXME see #24479 for reconsidering the stopping behaviour
-      Behaviors.stopped
-    }
-
   }
```

Comment:
We also have `onPersistRejected`. In untyped it was possible to override `def onPersistRejected`, but we don't have that hook here. Perhaps we should define a public exception and throw that one. Then one could define ordinary supervision to resume.

Reply:
I think the current behavior is dangerous as it just logs so users won't get back the event on next replay.

+1 to a publish exception